app/src/main/java/a3/com/convo/activities/LoginActivity.java

```diff
@@ -189,22 +221,39 @@ public void done(ParseException e) {
     }
 
     protected void getFriendsOnApp(AccessToken access_token) {
-        // TODO: add the friends' object IDs not facebook ids
         GraphRequest request = GraphRequest.newMyFriendsRequest(
                 access_token,
                 new GraphRequest.GraphJSONArrayCallback() {
                     @Override
                     public void onCompleted(JSONArray friends, GraphResponse response) {
                         try {
-                            ParseUser user = ParseUser.getCurrentUser();
+                            final ParseUser user = ParseUser.getCurrentUser();
                             // initialize empty likes array
-                            user.put("friends", new ArrayList<String>());
+                            user.put(FRIENDS, new ArrayList<String>());
                             for (int i = 0; i < friends.length(); i++) {
                                 JSONObject friend = friends.optJSONObject(i);
-                                String name = friend.optString("name");
-                                String id = friend.optString("id");
-                                user.add("friends", id);
-                                user.saveInBackground();
+                                String name = friend.optString(NAME);
+                                final String id = friend.optString(ID);
+                                ParseQuery<ParseUser> query = ParseUser.getQuery();
+                                query.whereEqualTo(USERNAME, id);
+                                query.findInBackground(new FindCallback<ParseUser>() {
+                                    @Override
+                                    public void done(List<ParseUser> objects, ParseException e) {
+                                        if (objects != null && objects.size() == 1) {
+                                            // get the friend ParseUser with the username matching the friend of current user
+                                            ParseUser friend = objects.get(0);
+                                            String objectId = friend.getObjectId();
```

Comment:
What if friend is null?

Reply:
Is it fine for me to simply add a log here? I didn't think it was possible for a Parse query to return an array with a null user when we never add a null user to our ParseUser class (even empty users are not supposed to be created).